frontend/src/categoryDetail.js

```diff
@@ -0,0 +1,51 @@
+import React, { Component } from 'react'
+
+class categoryDetail extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            bots: []
+        }
+    }
+
+    componentDidMount() {
+        const id = this.props.match.params.id;
+        fetch(`http://127.0.0.1:8000/api/category/${id}`).then(res => res.json()).then(
+            (result) => {
+                this.setState({
+                    bots: result.bots
+                });
+            }
+        );
+    }
+
+    componentDidUpdate(prevProps, prevState) {
```

Comment:
`prevState` not used, thus may be deleted
```suggestion
    componentDidUpdate(prevProps) {
```